jetstream/concepts/consumers.md

```diff
@@ -2,33 +2,115 @@
 Acknowledgements default to `AckExplicit` - the only supported mode for pull-based Consumers - meaning every message requires a distinct acknowledgement. But for push-based Consumers, you can set `AckNone` that does not require any acknowledgement, or `AckAll` which quite interestingly allows you to acknowledge a specific message, like message `100`, which will also acknowledge messages `1` through `99`. The `AckAll` mode can be a great performance boost.
 
-Some messages may cause your applications to crash and cause a never ending loop forever poisoning your system. The `MaxDeliver` setting allow you to set a upper bound to how many times a message may be delivered.
+Some messages may cause your applications to crash and cause a never ending loop forever poisoning your system. The `MaxDeliver` setting allow you to set an upper bound to how many times a message may be delivered.
 
 To assist with creating monitoring applications, one can set a `SampleFrequency` which is a percentage of messages for which the system should sample and create events. These events will include delivery counts and ack waits.
 
 When defining Consumers the items below make up the entire configuration of the Consumer:
 
-| Item | Description |
-| :--- | :--- |
-| AckPolicy | How messages should be acknowledged, `AckNone`, `AckAll` or `AckExplicit` |
-| AckWait | How long to allow messages to remain un-acknowledged before attempting redelivery |
-| DeliverPolicy | The initial starting mode of the consumer, `DeliverAll`, `DeliverLast`, `DeliverNew`, `DeliverByStartSequence` or `DeliverByStartTime` |
-| DeliverySubject | The subject to deliver observed messages. Useful to set up an alternate subject for a regular NatsSubcriber can listen on that subject. Not allowed for pull subscriptions. |
-| Durable | The name of the Consumer, which the server will track, allowing resuming consumption where left off. See [naming](../administration/naming.md). |
-| FilterSubject | When consuming from a Stream with many subjects, or wildcards, select only a specific incoming subjects, supports wildcards |
-| MaxDeliver | Maximum amount times a specific message will be delivered.  Use this to avoid poison pills crashing all your services forever |
-| OptStartSeq | When first consuming messages from the Stream start at this particular message in the set |
-| ReplayPolicy | How messages are sent `ReplayInstant` or `ReplayOriginal` |
-| SampleFrequency | What percentage of acknowledgements should be samples for observability, 0-100 |
-| OptStartTime | When first consuming messages from the Stream start with messages on or after this time |
-| RateLimit | The rate of message delivery in bits per second |
-| MaxAckPending | The maximum number of messages without acknowledgement that can be outstanding, once this limit is reached message delivery will be suspended |
+### AckPolicy
+How messages should be acknowledged. The server will consider an ack ony if it comes within the Ack Wait window.
```

Comment:
This seems bad. It's like hey there is a window, except when there isn't. Actually really really bad. Because if I ack after the window but another process is already consuming the same message, there would be multiple processes handling the same message. It's almost like the server should forcefully reject the late ack to let that consumer know. I'll try to document this.

Reply:
There would anyway be multiple processes if the first one overrun.

I agree with you though